pilot/pkg/networking/core/cluster_waypoint.go

```diff
@@ -90,12 +91,22 @@ func (configgen *ConfigGeneratorImpl) buildWaypointInboundClusters(
 }
 
 // `inbound-vip||hostname|port`. EDS routing to the internal listener for each pod in the VIP.
-func (cb *ClusterBuilder) buildWaypointInboundVIPCluster(svc *model.Service, port model.Port, subset string) *clusterWrapper {
+func (cb *ClusterBuilder) buildWaypointInboundVIPCluster(proxy *model.Proxy, svc *model.Service, port model.Port, subset string) *clusterWrapper {
 	clusterName := model.BuildSubsetKey(model.TrafficDirectionInboundVIP, subset, svc.Hostname, port.Port)
 
-	clusterType := cluster.Cluster_EDS
-	localCluster := cb.buildCluster(clusterName, clusterType, nil,
-		model.TrafficDirectionInbound, &port, nil, nil)
+	discoveryType := convertResolution(cb.proxyType, svc)
+	var lbEndpoints []*endpoint.LocalityLbEndpoints
+	if discoveryType == cluster.Cluster_STRICT_DNS || discoveryType == cluster.Cluster_LOGICAL_DNS {
+		lbEndpoints = endpoints.NewCDSEndpointBuilder(
+			proxy,
+			cb.req.Push,
+			clusterName,
+			model.TrafficDirectionOutbound, "", svc.Hostname, port.Port,
```

Comment:
Is this right changing from TrafficDirectionInbound to TrafficDirectionOutbound , and why

Reply:
I think this doesn't actually get used. let me double check and switch to inbound

Reply:
BTW the cluster name it is using inboundVIP

Reply:
>clusterName := model.BuildSubsetKey(model.TrafficDirectionInboundVIP

This is a legacy issue